### PR TITLE
Fix AutoNormal.quantiles

### LINKED ISSUE #2870: guide.quantiles() shape issue
I run into frequent `RuntimeError`s when generating quantiles of models fitted with SVI. This can be replicated on the radon models in `pyro-models`. For example, the [radon_intercept](https://github.com/pyro-ppl/pyro-models/blob/master/pyro_models/arm/radon_intercept_chr.py) model fit with `SVI` with an `AutoNormal` guide generates the following:

```
>>> estimates = guide.quantiles([0.05, 0.5, 0.95])


estimates = guide.quantiles([0.05, 0.5, 0.95])
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
~/GitHub/pyro-models/pyro_models/arm/radon_intercept_chr.py in 
----> 78 estimates = guide.quantiles([0.05, 0.5, 0.95])

~/miniforge3/envs/pyro/lib/python3.9/site-packages/torch/autograd/grad_mode.py in decorate_context(*args, **kwargs)
     25         def decorate_context(*args, **kwargs):
     26             with self.__class__():
---> 27                 return func(*args, **kwargs)
     28         return cast(F, decorate_context)
     29 

~/miniforge3/envs/pyro/lib/python3.9/site-packages/pyro/infer/autoguide/guides.py in quantiles(self, quantiles, *args, **kwargs)
    578 
    579             site_quantiles = torch.tensor(quantiles, dtype=site_loc.dtype, device=site_loc.device)
--> 580             site_quantiles_values = dist.Normal(site_loc, site_scale).icdf(site_quantiles)
    581             constrained_site_quantiles = biject_to(site["fn"].support)(site_quantiles_values)
    582             results[name] = constrained_site_quantiles

~/miniforge3/envs/pyro/lib/python3.9/site-packages/torch/distributions/normal.py in icdf(self, value)
     83 
     84     def icdf(self, value):
---> 85         return self.loc + self.scale * torch.erfinv(2 * value - 1) * math.sqrt(2)
     86 
     87     def entropy(self):

RuntimeError: The size of tensor a (85) must match the size of tensor b (3) at non-singleton dimension 0

```

This has come up repeatedly with larger models, but this is the first easily reproducible example that I have come across.

### Environment
For any bugs, please provide the following:
 - OS and python version. MacOS, Linux; Python 3.7.10
 - PyTorch version: 1.8.0
 - Pyro version: 1.6.0

### Code Snippet

Using the `arm.radon_intercept` model, I am fitting using the following:

```python
N_STEPS = 1000

guide = autoguides.AutoNormal(model)

initial_lr = 0.01
optimizer = pyro.optim.ClippedAdam({"lr": initial_lr})
svi = SVI(model, guide, optimizer, loss=Trace_ELBO())

pyro.clear_param_store()
for j in range(N_STEPS):
    # calculate the loss and take a gradient step
    loss = svi.step(radon_data)
    if not j % 100:
        print("[iteration %04d] loss: %.4f" % (j + 1, loss / len(radon_data['y'])))

estimates = guide.quantiles([0.05, 0.5, 0.95])
```

### PR DESCRIPTION
**Pull request by fehiepsi**
Resolves #2870

This fixes 2 issues of AutoNormal.quantiles:
+ quantiles assume the sites are scalars
+ output of quantiles is a tensor, rather than a list of tensors

### CODE REVIEW

File: pyro/infer/autoguide/guides.py [hunk 1]
```diff
@@ -574,9 +574,10 @@ def quantiles(self, quantiles, *args, **kwargs):
             site_loc, site_scale = self._get_loc_and_scale(name)
 
             site_quantiles = torch.tensor(quantiles, dtype=site_loc.dtype, device=site_loc.device)
+            site_quantiles = site_quantiles.reshape((-1,) + (1,) * site_loc.dim())
             site_quantiles_values = dist.Normal(site_loc, site_scale).icdf(site_quantiles)
             constrained_site_quantiles = biject_to(site["fn"].support)(site_quantiles_values)
-            results[name] = constrained_site_quantiles
+            results[name] = list(constrained_site_quantiles)
```
Comment (fritzo) [inline pyro/infer/autoguide/guides.py:580]: Can we avoid the unbind here?  I often use this method to return tensors.

**Comment by fritzo** [comment]
@fehiepsi I would like to keep the output of quantiles a dict of tensors, rather than a dict of lists of tensors.  Is there a particular reason you've changed that?

**Comment by fehiepsi** [comment]
I like to return a tensor, rather than a list of tensors. Should I go ahead to change docstring and the implementations in other AutoContinuous guides?

**Comment by fehiepsi** [comment]
@neerajprad I keep getting this coverall error. I guess currently, coveralls require some role?
```
Run GITHUB_SHA="$GITHUB_RUN_ID" GITHUB_REF="" coveralls
  GITHUB_SHA="$GITHUB_RUN_ID" GITHUB_REF="" coveralls
  shell: /usr/bin/bash -e {0}
  env:
    CXX: g++-8
    CC: gcc-8
    COVERALLS_PARALLEL: true
    COVERALLS_REPO_TOKEN: 
    pythonLocation: /opt/hostedtoolcache/Python/3.6.13/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.6.13/x64/lib
Running on Github Actions but GITHUB_TOKEN is not set. Add "env: GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}" to your step config.
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.6.13/x64/lib/python3.6/site-packages/coveralls/cli.py", line 66, in main
    base_dir=options.get('--basedir') or '')
  File "/opt/hostedtoolcache/Python/3.6.13/x64/lib/python3.6/site-packages/coveralls/api.py", line 47, in __init__
    self.ensure_token()
  File "/opt/hostedtoolcache/Python/3.6.13/x64/lib/python3.6/site-packages/coveralls/api.py", line 55, in ensure_token
    'Running on Github Actions but GITHUB_TOKEN is not set. '
coveralls.exception.CoverallsException: Running on Github Actions but GITHUB_TOKEN is not set. Add "env: GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}" to your step config.
Error: Process completed with exit code 1.
```

**Review by fritzo — APPROVED** [review]
Nice fix @fehiepsi!

**Comment by fehiepsi** [comment]
Our CI is still failing after the dev branch is updated, due to the same issue.

**Pull request merged**